### PR TITLE
[FIX] make travis work

### PR DESCRIPTION
**Pull request by hbrunn**
[IMP] try if sudo: false works

### CODE REVIEW

**Comment by hbrunn** [comment]
`python-ldap` is not (yet) supported, I requested to whitelist it in https://github.com/travis-ci/apt-package-whitelist/issues/312

**Comment by hbrunn** [comment]
travis is green!

**Comment by hbrunn** [comment]
@gurneyalex I see you pushed some changes yourself here, could you maybe review this PR? This uses the container infrastructure and gets rid of the partition-abominations

File: .travis.yml [hunk 1]
```diff
@@ -1,56 +1,40 @@
...
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^p[a-q]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^p[r-z]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^[q-r]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^s"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^[t-v]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^website_[a-m]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^website_[n-z]"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^w([^e]|(e[^b])|(eb[^s])|(ebs[^i])|(ebsi[^t])|(ebsit[^e])|(ebsite[^_]))"
-  - VERSION="8.0" ODOO_REPO="OCA/OCB" INCLUDE_REGEX="^[x-z]"
+  - VERSION="8.0" ODOO_REPO="OCA/OCB" LINT_CHECK="0" INSTALL_OPTIONS="--init base"
```
Comment (dreispt) [inline .travis.yml]: I guess that with https://github.com/OCA/maintainer-quality-tools/pull/212 the `--init base` is no longer needed

**Comment by gurneyalex** [comment]
@hbrunn can you rebase so we can get the `cache: pip` line I recently added (and a mergeable PR)?

by code review only I'm :+1:

File: .travis.yml (hunk 1 above)
Comment (hbrunn) [inline .travis.yml]: @dreispt something seems to be wrong here: https://travis-ci.org/OCA/OCB/builds/73141859

**Comment by dreispt** [comment]
Ouch! If failing on my function `get_test_dependencies()`

**Comment by hbrunn** [comment]
The issue here is that we have to look at two addon paths, $repo/addons and $repo/openerp/addons. I solved that somewhat hacky by setting TRAVIS_BUILD_DIR to the latter.
You probably should call get_test_dependencies with the result of get_addons_path, split that again and search for the manifests in all of them

**Comment by dreispt** [comment]
Why do we need two addons paths? For OCB can't we skip the Odoo server clone and directly use the repo under tests? It already contains all the addons.

**Comment by hbrunn** [comment]
I don't think I understand exactly what you mean, but you have the core addons and their tests at /addons, but also a lot of api testing code in /openerp/addons/test_*: https://github.com/OCA/OCB/tree/8.0/openerp/addons, so I think we need both indeed

**Comment by hbrunn** [comment]
@dreispt I use a branch with my proposed fix now, and it seems to work now. Could you look into my proposal?

@all please don't merge this until the mtq branch is reverted to the normal one

**Comment by hbrunn** [comment]
green :v:

**Comment by gurneyalex** [comment]
@hbrunn thanks!

**Comment by StefanRijnhart** [comment]
You replace the wkhtmltopdf download with the Travis built-in and then disable report_webkit because of the X session error. However, typically this wkhtmltopdf error does not occur with the gna.org version. Should this be reverted?

**Comment by hbrunn** [comment]
I think ignoring `report_webkit` for now is an acceptable tradeoff for the speed we gain with the docker infrastructure. Pulling wkhtmltopdf from somewhere else and reenabling testing `report_webkit` would be a second step imho

**Comment by StefanRijnhart** [comment]
Thanks for the explanation! Fine by me :+1:

**Comment by pedrobaeza** [comment]
Thanks for the work

:+1: when Travis finishes

**Comment by hbrunn** [comment]
@StefanRijnhart  I reenabled testing `report_webkit` after chekcing that travis' version of `wkhtmltopdf` works, it just needs an X server

**Comment by StefanRijnhart** [comment]
I see, you enabled an Xserver in the travis session. Perfect!

**Comment by gurneyalex** [comment]
I'll make PRs on other OCA branches still using sudo for whtmltopdf. Thanks @hbrunn for showing the way.

**Comment by StefanRijnhart** [comment]
@gurneyalex can I take your comment as an approval?

**Comment by pedrobaeza** [comment]
I think so, because he has already used it in other repo, so you can merge when you want.

**Comment by gurneyalex** [comment]
@StefanRijnhart yes.

**Comment by StefanRijnhart** [comment]
@gurneyalex thanks ;-)

**Pull request merged**